cli/cli.go

```diff
@@ -26,17 +26,27 @@ func Run() {
 			Usage:  "debug mode",
 			EnvVar: "DEBUG",
 		},
-
 		cli.StringFlag{
 			Name:  "log-level, l",
 			Value: "info",
 			Usage: fmt.Sprintf("Log level (options: debug, info, warn, error, fatal, panic)"),
 		},
+		cli.StringFlag{
+			Name:  "log-file",
+			Usage: "Logging the output to a file instead of stdout",
+		},
 	}
-
 	// logs
 	app.Before = func(c *cli.Context) error {
-		log.SetOutput(os.Stderr)
+		if c.IsSet("log-file") {
+			logDir := c.String("log-file")
+			fmt.Print(logDir)
+			file, err := os.OpenFile(logDir, os.O_RDWR, 0777)
+			fmt.Print(err)
```

Comment:
Instead of printing errors to the console, we should `log.SetOutput(os.Stderr)` before the `if c.IsSet("log-file") {` and use `log.Fatalf` when issuing `--log-file` related errors.

Also, no need to have a `else {` in that case